src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderExtensions.cs

```diff
@@ -278,11 +278,36 @@ private static bool IsTypeParameterEquivalentToTypeInst(this Type typeParam, Typ
 
         public static bool HasSameMetadataDefinitionAs(this MemberInfo mi1, MemberInfo mi2)
         {
+            if (s_lazyHasSameMetadataDefinitionAsApi == null)
+            {
+                MethodInfo apiMethod = typeof(MemberInfo).GetMethod(
```

Comment:
I'd expect it to be something like:

```C#
try
{
         MethodInfo apiMethod = typeof(MemberInfo).GetMethod(
                    "HasSameMetadataDefinitionAs",
                    BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly | BindingFlags.ExactBinding,
                    binder: null,
                    types: new Type[] { typeof(MemberInfo) },
                    modifiers: null);

        if (apiMethod != null)
        {
               var sameDefAs  = (Func<MemberInfo, MemberInfo, bool>)Delegate.CreateDelegate(typeof(Func<MemberInfo, MemberInfo, bool>), apiMethod);

               var result = sameDefAs  (m1, m2);

               // it worked, so publish it
               s_MemberEquivalence = sameDefAs;
               return result;
        }
}
catch
{
          // Platform might not allow access to the API or it is not implemented.
}
```



---
In reply to: [121757257](https://github.com/dotnet/corefx/pull/20991#discussion_r121757257) [](ancestors = 121757257,121757009)